src/Bridge/Graphql/Type/SchemaBuilder.php

```diff
@@ -0,0 +1,237 @@
+            'name' => $shortName,
+            'description' => $resourceMetadata->getDescription(),
+            'fields' => function () use ($resource) {
+                return $this->getResourceTypeFields($resource);
+            },
+        ]);
+    }
+
+    /**
+     * Gets the fields of the type of the given resource.
+     */
+    private function getResourceTypeFields(string $resource): array
+    {
+        $fields = [];
+
+        foreach ($this->propertyNameCollectionFactory->create($resource) as $property) {
+            $propertyMetadata = $this->propertyMetadataFactory->create($resource, $property);
+
+            $propertyType = $propertyMetadata->getType();
+            if (null !== $propertyType) {
+                try {
+                    $fields[$property] = [
+                        'type' => $this->convertType($propertyType),
+                        'description' => $propertyMetadata->getDescription(),
+                    ];
+                } catch (InvalidTypeException $e) {
+                    continue;
+                }
+            }
+        }
+
+        return $fields;
+    }
+
+    /**
+     * Gets the field arguments of the identifier of a given resource.
+     */
+    private function getResourceIdentifierFieldArguments(string $resource): array
+    {
+        foreach ($this->propertyNameCollectionFactory->create($resource) as $property) {
+            $propertyMetadata = $this->propertyMetadataFactory->create($resource, $property);
+            $propertyIsIdentifier = $propertyMetadata->isIdentifier() ?? false;
+            $propertyType = $propertyMetadata->getType();
+
+            if (!$propertyIsIdentifier || null === $propertyType) {
+                continue;
+            }
+
+            // @TODO Manage multiple identifiers?
+            return [
+                'name' => $property,
+                'type' => GraphQLType::nonNull($this->convertType($propertyType)),
+                'description' => $propertyMetadata->getDescription(),
+            ];
+        }
+
+        throw new \LogicException(sprintf('Missing identifier field for resource "%s"', $resource));
+    }
+
+    /**
+     * Converts a built-in type to its GraphQL equivalent.
+     */
+    private function convertType(Type $type): GraphQLType
+    {
+        switch ($type->getBuiltinType()) {
+            case Type::BUILTIN_TYPE_BOOL:
+                $graphqlType = GraphQLType::boolean();
+                break;
+            case Type::BUILTIN_TYPE_INT:
+                $graphqlType = GraphQLType::int();
+                break;
+            case Type::BUILTIN_TYPE_FLOAT:
+                $graphqlType = GraphQLType::float();
+                break;
+            case Type::BUILTIN_TYPE_STRING:
+                $graphqlType = GraphQLType::string();
+                break;
+            case Type::BUILTIN_TYPE_OBJECT:
+                if (\DateTime::class === $type->getClassName()) {
```

Comment:
```php
if (is_subclass_of($type->getClassName(), \DateTimeInterface::class)) {
```

Reply:
We can even nitpick with `is_a($type->getClassName(), \DateTimeInterface::class, true)`